### PR TITLE
fix: Add name string legitimacy check.

### PR DESCRIPTION
**Pull request by zhongwencool**
-  Add user/appid, name only allow `^[A-Za-z0-9]+[A-Za-z0-9-_]*$`
Adding a user with `/` will cause the HTTP API to fail to delete the user.

- I added a comment to `emqx_dashboard.conf` file and also **alerted on startup**
If there is an admin account in the configuration file, the log will prompt the user to delete something in the configuration file at startup.
```erlang
Start http:management listener on 8081 successfully.
2022-03-16T11:02:05.793717+08:00 [warning] [Dashboard] The initial default password for dashboard 'admin' 
user in emqx_dashboard.conf, For safety, it should be changed as soon as possible., 
Please use the './bin/emqx_ctl admins' CLI to change it. 
Then remove `dashboard.default_user.login/password` from emqx_dashboard.conf
Start http:dashboard listener on 18083 successfully.
EMQ X Broker 4.3.13-rc.1-b44512cd is running now!
Eshell V12.0.2  (abort with ^G)
(emqx@127.0.0.1)1>
```

In order to avoid using the command line to delete admin/public, which was in effect at the time, after reboot, admin/public will be rewritten into the system again.

### CODE REVIEW

**Comment by zmstone** [comment]
possible to delete bad IDs from CLI?

**Comment by zmstone** [comment]
> possible to delete bad IDs from CLI?

If already created

File: src/emqx_misc.erl [hunk 1]
```diff
@@ -52,6 +52,26 @@
         , hexstr2bin/1
         ]).
 
+-export([ valid_str/1
+        ]).
+
+-define(VALID_STR_RE, "^[A-Za-z]+[A-Za-z0-9-_]*$").
+
+-spec valid_str(list() | binary()) -> ok | {error, Reason::binary()}.
+valid_str(Str) ->
```
Comment (zmstone) [inline src/emqx_misc.erl:61]:
```suggestion
is_sane_id(Str) ->
```

**Comment by zhongwencool** [comment]
> > possible to delete bad IDs from CLI?
> 
> If already created

Don't delete automatically, just remind the user to change.

2022-03-16T15:18:21.072064+08:00 [warning] [app] /3/e/3e7dbb6dd8c is not a sane appid(^[A-Za-z]+[A-Za-z0-9-_]*$). Please use `emqx_ctl mgmt delete /3/e/3e7dbb6dd8c` to delete it and create a new one.
Start http:management listener on 8081 successfully.
2022-03-16T15:18:21.101644+08:00 [warning] [Dashboard] The initial default password for dashboard 'admin' user in emqx_dashboard.conf, For safety, it should be changed as soon as possible., Please use the './bin/emqx_ctl admins' CLI to change it., Then remove `dashboard.default_user.login/password` from emqx_dashboard.conf
2022-03-16T15:18:21.102197+08:00 [warning] [dashboard] `admin/test` is not a sane username(^[A-Za-z0-9]+[A-Za-z0-9-_]*$). Please use `emqx_ctl admins del admin/test` to delete it and create a new one.
Start http:dashboard listener on 18083 successfully.
EMQ X Broker 4.3.13-rc.1-f70d7776 is running now!
Eshell V12.0.2  (abort with ^G)

**Comment by zhongwencool** [comment]
we should port this to :https://github.com/emqx/emqx-enterprise/pull/816

**Review by DDDHuang — APPROVED** [review]

**Review by zmstone — COMMENTED** [review]
nit: some of the commits are no longer valid, consider squash

**Comment by zhongwencool** [comment]
> nit: some of the commits are no longer valid, consider squash

We're in a hurry to release 4.4.2, :) 
need sync this to 4.4

**Pull request merged**